src/screens/DiscoveryScreenV2/hooks/useDiscovery.js

```diff
@@ -57,10 +58,13 @@ const useDiscovery = () => {
   };
 
   const topics = React.useMemo(() => {
-    return discovery.initialTopics.map((item) => ({
-      ...item,
-      following: item.following !== undefined ? item.following : item.user_id_follower !== null
-    }));
+    return uniqBy(
+      discovery.initialTopics.map((item) => ({
+        ...item,
+        following: item.following !== undefined ? item.following : item.user_id_follower !== null
+      })),
+      (item) => item.topic_id
+    );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The use of `uniqBy` here helps to remove duplicates based on `topic_id`, which is a good practice to avoid redundant data. However, please ensure that there are no side effects due to the removal of duplicate topics. For instance, if there are any functionalities depending on the count or order of the original `discovery.initialTopics` array, those might be affected.
```diff
+ return uniqBy(
+   discovery.initialTopics.map((item) => ({
+     ...item,
+     following: item.following !== undefined ? item.following : item.user_id_follower !== null
+   })),
+   (item) => item.topic_id
+ );
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->